upload/library/Sidane/Threadmarks/Install.php

```diff
@@ -2,37 +2,50 @@
 
 class Sidane_Threadmarks_Install
 {
-  public static function install($existingAddOn, $addOnData)
-  {
-    if ($existingAddOn['version_id'] > 1) {
-      return;
-    }
+    public static function install($existingAddOn, $addOnData)
+    {
+        $version = isset($existingAddOn['version_id']) ? $existingAddOn['version_id'] : 0;
+        
+        $db = XenForo_Application::get('db');
+        $tables_created = false;
 
-    $db = XenForo_Application::get('db');
+        if ($version == 0)
+        {
+            $db->query("
+              CREATE TABLE IF NOT EXISTS threadmarks (
+                threadmark_id INT UNSIGNED NOT NULL PRIMARY KEY AUTO_INCREMENT,
+                thread_id INT UNSIGNED NOT NULL,
+                post_id INT UNSIGNED NOT NULL,
+                label VARCHAR(100) NOT NULL,
+                UNIQUE KEY `thread_post_id` (`thread_id`,`post_id`)
+              ) ENGINE = InnoDB CHARACTER SET utf8 COLLATE utf8_general_ci
+            ");
+            $tables_created = true;
+        }
 
-    $db->query("
-      CREATE TABLE IF NOT EXISTS threadmarks (
-        threadmark_id INT UNSIGNED NOT NULL PRIMARY KEY AUTO_INCREMENT,
-        thread_id INT UNSIGNED NOT NULL,
-        post_id INT UNSIGNED NOT NULL,
-        label VARCHAR(100) NOT NULL,
-        KEY thread_id (thread_id)
-      ) ENGINE = InnoDB CHARACTER SET utf8 COLLATE utf8_general_ci
-    ");
+        if ($version == 1)
+        {
+            $db->query("ALTER TABLE xf_thread DROP COLUMN has_threadmarks");
+            $db->query("ALTER TABLE xf_thread DROP INDEX thread_id ");
```

Comment:
I don't think this should be here. Add-on upgrade broke when it reached this line.